admin-dev/themes/new-theme/js/pages/order/view/order-product-add.js

```diff
@@ -0,0 +1,127 @@
+      this.priceTaxIncludedInput.val($(event.currentTarget).find(':selected').attr('data-price-tax-included'));
+      this.available = $(event.currentTarget).find(':selected').attr('data-stock');
+      this.quantityInput.trigger('change');
+    });
+    this.quantityInput.on('change keyup', (event) => {
+      if (this.available === null) {
+        return;
+      }
+      const quantity = parseInt(event.target.value ? event.target.value : 0, 10);
+      const available = this.available - quantity;
+      this.availableText.text(available);
+      this.availableText.toggleClass('text-danger font-weight-bold', available < 0);
+    });
+    this.productIdInput.on('change', () => {
+      this.productAddActionBtn.removeAttr('disabled');
+    });
+    this.priceTaxIncludedInput.on('change keyup', (event) => {
+      let priceTaxIncl = parseFloat(event.target.value);
+      if (priceTaxIncl < 0 || isNaN(priceTaxIncl)) {
+        priceTaxIncl = 0;
+      }
+      const taxRate = this.taxRateInput.val() / 100 + 1;
+      this.priceTaxExcludedInput.val(ps_round(priceTaxIncl / taxRate, 2));
+    });
+    this.priceTaxExcludedInput.on('change keyup', (event) => {
+      let priceTaxExcl = parseFloat(event.target.value);
+      if (priceTaxExcl < 0 || isNaN(priceTaxExcl)) {
+        priceTaxExcl = 0;
+      }
+      const taxRate = this.taxRateInput.val() / 100 + 1;
+      this.priceTaxIncludedInput.val(ps_round(priceTaxExcl * taxRate, 2));
+    });
+    this.productAddActionBtn.on('click', event => this.addProduct($(event.currentTarget).data('order-id')));
+  }
+
+  setProduct(product) {
+    this.productIdInput.val(product.product_id).trigger('change');
+    this.priceTaxExcludedInput.val(product.price_tax_excl);
+    this.priceTaxIncludedInput.val(product.price_tax_incl);
+    this.taxRateInput.val(product.tax_rate);
+    this.available = product.stock;
+    this.quantityInput.val(1);
+    this.quantityInput.trigger('change');
+    this.setCombinations(product.combinations);
+  }
+
+  setCombinations(combinations) {
+    this.combinationsSelect.empty();
+    Object.entries(combinations).forEach((val) => {
+      this.combinationsSelect.append(`<option value="${val[1].attribute_combination_id}" data-price-tax-excluded="${val[1].price_tax_excluded}" data-price-tax-included="${val[1].price_tax_included}" data-stock="${val[1].stock}">${val[1].attribute}</option>`);
+    });
+    this.combinationsBlock.toggleClass('d-none', Object.keys(combinations).length === 0);
+  }
+
+  addProduct(orderId) {
+    const params = {
+      product_id: this.productIdInput.val(),
+      combination_id: $(':selected', this.combinationsSelect).val(),
+      price_tax_incl: this.priceTaxIncludedInput.val(),
+      price_tax_excl: this.priceTaxExcludedInput.val(),
+      quantity: this.quantityInput.val(),
+    };
+    $.ajax({
+      url: this.router.generate('admin_orders_add_product', {orderId}),
+      method: 'POST',
+      data: params,
+    }).then((response) => {
```

Comment:
Can we handle the error case please ? 🙏 just display an error message to tell the user something bad happened

Reply:
:ballot_box_with_check: Thanks